CacheManager.StackExchange.Redis.SelfCleaning.Tests/SelfCleaningRedisIntegrationTests.cs

```diff
@@ -0,0 +1,247 @@
+            {
+                RedisValue[] newValues = {values[0], values[2], values[3], values[4], values[1], values[5], true};
+                (RedisValue[], DateTime) tuple = (newValues, DateTime.Now);
+
+                RedisKey key = keys[0];
+
+                if (_fauxDatabase.ContainsKey(key))
+                {
+                    _fauxDatabase[key] = tuple;
+                }
+                else
+                {
+                    _fauxDatabase.Add(key, tuple);
+                }
+            }
+
+            // Put
+            _databaseMock
+                .Setup(database => database.ScriptEvaluate(It.IsAny<byte[]>(), It.IsAny<RedisKey[]>(),
+                    It.Is<RedisValue[]>(values => values != null), It.IsAny<CommandFlags>()))
+                .Callback<byte[], RedisKey[], RedisValue[], CommandFlags>(Put);
+
+            // Get
+            _databaseMock
+                .Setup(database => database.ScriptEvaluate(It.IsAny<byte[]>(), It.IsAny<RedisKey[]>(),
+                    It.Is<RedisValue[]>(values => values == null), It.IsAny<CommandFlags>()))
+                .Returns<byte[], RedisKey[], RedisValue[], CommandFlags>((hash, keys, values, flags) =>
+                    RedisResult.Create(_fauxDatabase[keys[0]].Values));
+
+            // Remove
+            _databaseMock
+                .Setup(database => database.KeyDelete(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+                .Callback<RedisKey, CommandFlags>((key, flags) => _fauxDatabase.Remove(key));
+        }
+
+        private void SetupConnectionMock()
+        {
+            _connectionMock.SetupGet(connection => connection.Configuration).Returns("connectionString");
+            _connectionMock.Setup(connection => connection.GetEndPoints(It.IsAny<bool>())).Returns(new EndPoint[1]);
+            _connectionMock.Setup(connection => connection.GetServer(It.IsAny<EndPoint>(), It.IsAny<object>()))
+                .Returns(_serverMock.Object);
+            _connectionMock.Setup(connection => connection.GetDatabase(It.IsAny<int>(), It.IsAny<object>()))
+                .Returns(_databaseMock.Object);
+        }
+
+        private void SetupCache()
+        {
+            double cleanupInterval = 100;
+            var cleanupTimer = new DefaultTimer(cleanupInterval);
+            _timeToLive = TimeSpan.FromSeconds(1);
+
+            _cache = CacheFactory.Build<DummyModel>(part => part
+                .WithProtoBufSerializer()
+                .WithSelfCleaningRedisConfiguration(_connectionMock.Object, cleanupTimer, _timeToLive,
+                    out string configurationKey)
+                .WithSelfCleaningRedisCacheHandle(configurationKey));
+
+            _onRemoveByHandleInvocations = new List<(CacheItemRemovedEventArgs, DateTime)>();
+            _cache.OnRemoveByHandle += (sender, args) => _onRemoveByHandleInvocations.Add((args, DateTime.Now));
+
+            _cache.CacheHandles.OfType<IStartable>().Single().Start();
```

Comment:
Isnt that better? 
![image](https://user-images.githubusercontent.com/62211836/78126673-8d319f80-741b-11ea-8774-33eb386ab4f4.png)

Reply:
changed